context_test.go

```diff
@@ -208,25 +194,12 @@ func TestContextWithHeadersAsContext(t *testing.T) {
 
 func TestContextBuilderParentContextSpan(t *testing.T) {
 	ctx := getParentContext(t)
-	span := NewSpan(5, 4, 3)
+	assert.Equal(t, "some value", ctx.Value("some key"))
 
 	ctx2, _ := NewContextBuilder(time.Second).
 		SetParentContext(ctx).
-		SetSpanForTest(&span).
-		Build()
-	assert.Equal(t, &span, CurrentSpan(ctx2), "explicitly provided span used")
-
-	ctx3, _ := NewContextBuilder(time.Second).
-		SetParentContext(ctx2).
-		Build()
-	assert.Equal(t, &span, CurrentSpan(ctx3), "span inherited from parent")
-
-	ctx4, _ := NewContextBuilder(time.Second).
-		SetParentContext(ctx2).
-		SetExternalSpan(3, 2, 1, true).
 		Build()
-	span4 := NewSpan(3, 2, 1)
-	assert.Equal(t, &span4, CurrentSpan(ctx4), "external span used")
+	assert.Equal(t, "some value", ctx2.Value("some key"), "key/value propagated from parent ctx")
 
 	goroutines.VerifyNoLeaks(t, nil)
 }
```

Comment:
Can we add a test where you create a span outside of tchannel using opentracing, pass that context in, and call `CurrentSpan` and validate it matches?